dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/servicesmanager/ServicesManagerStaticAnchor.java

```diff
@@ -10,65 +10,70 @@
  *******************************************************************************/
 package com.ibm.jbatch.container.servicesmanager;
 
+import java.security.AccessController;
+
+import org.osgi.framework.Bundle;
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.FrameworkUtil;
 import org.osgi.framework.ServiceReference;
 import org.osgi.util.tracker.ServiceTracker;
 import org.osgi.util.tracker.ServiceTrackerCustomizer;
 
+import com.ibm.ws.kernel.service.util.SecureAction;
+
 /**
  * There are various parts of the batch runtime that need to obtain refs
  * to batch OSGi services statically, outside of the normal OSGI service
  * component lifecycle and dependency management mechanisms.
- * 
+ *
  * For example, BatchWorkUnit is a Runnable that gets scheduled via Executors.
  * Once it's running it needs access to various batch service components.
- * We could manually "inject" the needed services into the BatchWorkUnit 
- * (pass them into the CTOR, e.g).  However there's no guarantee the work will
+ * We could manually "inject" the needed services into the BatchWorkUnit
+ * (pass them into the CTOR, e.g). However there's no guarantee the work will
  * run any time soon, and it's unsafe to hold onto service refs for a long
- * time since they may get deactivated or replaced. 
- * 
- * There's also the possibility in the future that work will be scheduled 
+ * time since they may get deactivated or replaced.
+ *
+ * There's also the possibility in the future that work will be scheduled
  * via persistent schedulers, or possibly via (hypothetical) "distributed" schedulers
  * that schedule the work to other JVMs (think - the batch manager scheduling
- * to another endpoint within the collective).  These scenarios further complicate
+ * to another endpoint within the collective). These scenarios further complicate
  * the matter.
- * 
+ *
  * So while good design would have us avoid static at all costs, in this case
- * we're kind of stuck with it for now (barring significant refactoring of code). 
- * 
+ * we're kind of stuck with it for now (barring significant refactoring of code).
+ *
  */
 public class ServicesManagerStaticAnchor {
 
     /**
      * Static instance of the ServicesManager service object.
      */
     private static ServicesManager servicesManagerStaticInstance;
-    
+
     /**
      * ServiceTracker singleton tracking the ServicesManager.
      */
     private static volatile ServiceTracker<ServicesManager, ServicesManager> serviceTrackerSingleton;
 
+    protected final static SecureAction priv = AccessController.doPrivileged(SecureAction.get());
```

Comment:
I'm curious what this is needed for.. are there times we need a privileged action to get the bundle context?

Reply:
It's always a privileged action.. the "difference" is that the Open Liberty FATs run by default with j2sec enabled, unlike other environments.

Reply:
The other thing that's "new" is that the previous test only used memory persistence (though actually I changed in this PR so that test use JDBC persistence now and have the new tran timeout test using memory persistence because it's quicker).    I forget the exact code path but `ServicesManagerStaticAnchor` was needed because of the different threading involved in (I think it was) loading from the DB.